test/functional/test-crypto.js

```diff
@@ -159,128 +159,98 @@ describes.realWin('crypto-impl', {}, env => {
-        'da7675333e753d0a5de96c9a5fc052319134b32dc535a31f51382ddc24b946ac' +
-        '6134ea900af5a00ab7a528486f03273814b61207de1609864901f85f4f83ee74' +
-        '723802e9c2c7eacf3d1c2f8c6b5d4295b6ad3166a0c9f7482d494c9b8723add6' +
-        '2ce8d5868378c720a23cec6b80f802d1021323e3d22a57551de554d292d5fd64' +
-        '4de94fdb376b3e689e4d6239dac65bc4e9ae3265b10bfe87bb838aa15515b01e' +
-        '1cf21c14f99d474e12f5fcbeba8697074176d9bf8e3463d6195805c883710299' +
-        'e50f0eef74');
-    const data = stringToBytes('Hello');
-    const wrongData = stringToBytes('Hello0');
-
-    let win;
-    let crypto;
-    let pubKeyInfoPromise;
-    let pubKeyInfoPromise1;
-
-    beforeEach(() => {
-      crypto = createCrypto(win || env.win);
-      pubKeyInfoPromise = crypto.importPublicKey(serviceName, {
-        kty: 'RSA',
-        'n': modulus,
-        'e': pubExp,
-        alg: 'RS256',
-        ext: true,
-      });
-      pubKeyInfoPromise1 = crypto.importPublicKey(serviceName, {
-        kty: 'RSA',
-        'n': modulus1,
-        'e': pubExp1,
-        alg: 'RS256',
-        ext: true,
-      });
-    });
-
-    it('should resolve to a PublicKeyInfoDef object', () => {
-      if (!crypto.isCryptoAvailable()) { return; }
-      return pubKeyInfoPromise.then(pubKeyInfo => {
-        expect(pubKeyInfo).to.not.be.null;
-        expect(pubKeyInfo.serviceName).to.equal(serviceName);
-        expect(pubKeyInfo.hash).to.not.be.null;
-        expect(pubKeyInfo.hash.length).to.equal(4);
-        expect(pubKeyInfo.cryptoKey).to.not.be.null;
-      });
-    });
-
-    describe('verifySignature', function() {
-
-      it('should validate with the correct key and signature', () => {
-        if (!crypto.isCryptoAvailable()) { return; }
-        return pubKeyInfoPromise.then(pubKeyInfo =>
-            crypto.verifySignature(data, signature, pubKeyInfo)
-            .then(isvalid => expect(isvalid).to.be.true));
+      beforeEach(() => {
+        crypto = createCrypto(win || env.win);
       });
 
-      it('should not validate with the correct key but wrong data', () => {
-        if (!crypto.isCryptoAvailable()) { return; }
-        // Test with correct key, but wrong data.
-        return pubKeyInfoPromise.then(pubKeyInfo =>
-            crypto.verifySignature(wrongData, signature, pubKeyInfo)
-            .then(isvalid => expect(isvalid).to.be.false));
-      });
+      it('should create a well-formed RSA JSON Web Key (this tests the test)',
+         () => {
+           return publicJwkPromise.then(jwk => {
+             expect(jwk['kty']).to.equal('RSA');
+             expect(jwk['e']).to.equal('AQAB');
+             expect(jwk['n']).to.have.lengthOf(342);
+           });
+         });
+
+      describe('verifyPkcs', function() {
+
+        it('should validate with the correct key and signature', () => {
+          return publicJwkPromise.then(jwk => crypto.importPkcsKey(jwk))
+              .then(
+                  key => signaturePromise.then(
```

Comment:
Nit: Here and elsewhere in this file, we can move the callbacks to the same line as .then.